contrib/udfs/src/main/java/org/apache/drill/exec/udfs/NearestDateUtils.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.udfs;
+
+public class NearestDateUtils {
+  /**
+   * Specifies the time grouping to be used with the nearest date function
+   */
+  private enum TimeInterval {
+    YEAR,
+    QUARTER,
+    MONTH,
+    WEEK_SUNDAY,
+    WEEK_MONDAY,
+    DAY,
+    HOUR,
+    HALF_HOUR,
+    QUARTER_HOUR,
+    MINUTE,
+    HALF_MINUTE,
+    QUARTER_MINUTE,
+    SECOND
+  }
+
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(NearestDateUtils.class);
+
+  /**
+   * This function takes a Java LocalDateTime object, and an interval string and returns
+   * the nearest date closets to that time.  For instance, if you specified the date as 2018-05-04 and YEAR, the function
+   * will return 2018-01-01
+   *
+   * @param d the original datetime before adjustments
+   * @param  interval The interval string to deduct from the supplied date
+   * @return the modified LocalDateTime
+   */
+  public final static java.time.LocalDateTime getDate(java.time.LocalDateTime d, String interval) {
+    java.time.LocalDateTime newDate = d;
+    int year = d.getYear();
+    int month = d.getMonth().getValue();
+    int day = d.getDayOfMonth();
+    int hour = d.getHour();
+    int minute = d.getMinute();
+    int second = d.getSecond();
+    TimeInterval adjustmentAmount;
+    try {
+      adjustmentAmount = TimeInterval.valueOf(interval.toUpperCase());
+    } catch (Exception e) {
+      throw new org.apache.drill.common.exceptions.DrillRuntimeException(interval + " is not a valid time statement.");
+    }
```

Comment:
1. Use import rather than full path.
2. Please add more details to the exception message (for details please refer to my comments in previous rounds of code review, I wrote about this twice).

Reply:
Fixed the import statement.  I haven't done the error messages yet.

Reply:
Fixed